libobs/obs-nix.c

```diff
@@ -47,24 +47,24 @@ const char *get_module_extension(void)
 	return ".so";
 }
 
-#ifdef __LP64__
-#define BIT_STRING "64bit"
-#else
-#define BIT_STRING "32bit"
-#endif
-
 #define FLATPAK_PLUGIN_PATH "/app/plugins"
 
 static const char *module_bin[] = {
-	OBS_INSTALL_PREFIX "/" OBS_PLUGIN_DESTINATION,
-	"../../obs-plugins/" BIT_STRING,
-	FLATPAK_PLUGIN_PATH "/" OBS_PLUGIN_DESTINATION,
+	OBS_PLUGIN_PATH, FLATPAK_PLUGIN_PATH "/" OBS_PLUGIN_DESTINATION,
+	"../../obs-plugins/64bit",
+#ifdef OBS_INSTALL_PREFIX
+	OBS_INSTALL_PREFIX "/" OBS_PLUGIN_DESTINATION
+#endif
```

Comment:
The order matters mainly for two purposes (which might be opposing depending on use case):

* When optimising for speed, the most common entries should come first so plugins are found "early"
* When optimising for customisability, the most common entry should come last (so custom installations can come first

Because the same code is used for first-party runtime modules as well as plugins, we're at a bit of a crossroads: The former should never be "overridden", for the latter I haven't heard a compelling use case for yet.

Reply:
I mostly ask because the order is changed with this change.

Reply:
The change as-is optimises for speed (default locations first):

1. Default plugin path (does not use architecture subdirectory)
2. Flatpak custom path
3. Path relative to binary (uses legacy architecture subdirectory)
4. Absolute custom install prefix path

The default paths are relative to the OBS binary, using the distribution specific subdirectory names for `lib`, `share`, etc. so we're not reliant on absolute paths there (which is the main difference to variant 4).

Reply:
Sure. I just wanted to confirm that it _is_ a behavior change, in case anyone is concerned about that, or we start seeing weird reports afterward.

Reply:
Resolving this. No one else has raised this as a concern, so I'll let this be, and we'll just have to remember that this was changed.